components/camel-stream/src/main/java/org/apache/camel/component/stream/StreamEndpoint.java

```diff
@@ -303,7 +303,8 @@ public void setReadTimeout(int readTimeout) {
     //-------------------------------------------------------------------------
 
     @Override
-    protected void doStart() throws Exception {
+    protected void doInit() throws Exception {
+        super.doInit();
```

Comment:
charsets are potentially not available if native compiling, so we may consider to make this doStart. However there are ways to make native compilation include all charsets.